src/cli_common.c

```diff
@@ -192,3 +197,61 @@ int cliSecureInit()
 #endif
     return REDIS_OK;
 }
+
+sds readArgFromStdin(void) {
+    char buf[1024];
+    sds arg = sdsempty();
+
+    while(1) {
+        int nread = read(fileno(stdin),buf,1024);
+
+        if (nread == 0) break;
+        else if (nread == -1) {
+            perror("Reading from standard input");
+            exit(1);
+        }
+        arg = sdscatlen(arg,buf,nread);
+    }
+    return arg;
+}
+
+/* Create an sds array from argv, either as-is or by dequoting every
+ * element. When quoted is non-zero, may return a NULL to indicate an
+ * invalid quoted string.
+ */
```

Comment:
Let's document that the return value should be freed with `sdsfreesplitres` (although really awkward), or maybe create a thin wrapper, and document that that's the one that should be used.